cognite_toolkit/_cdf_tk/warnings.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import annotations
+
+import dataclasses
+import itertools
+from abc import ABC, abstractmethod
+from collections import UserList
+from collections.abc import Collection
+from dataclasses import dataclass
+from enum import Enum
+from functools import total_ordering
+from typing import Any, ClassVar, Generic, TypeVar, Union
+
+RICH_WARNING_FORMAT = "    [bold yellow]WARNING:[/] "
+RICH_WARNING_DETAIL_FORMAT = f"{'    ' * 2}"
+
+
+class SeverityLevel(Enum):
+    HIGH = "HIGH"
+    MEDIUM = "MEDIUM"
+    LOW = "LOW"
+
+
+class SeverityFormat:
+    @staticmethod
+    def get_rich_severity_format(severity: SeverityLevel, *messages: str) -> str:
```

Comment:
I am a bit skeptical about this one. I think my main issues are the following
* You have to pass in severity.
* Messages are enforced to be given in order. 

But I am not entirely convinced that I am right, so will approve and lets start using it. If it turns out to be error prone we can always change this mechanism.